### PR TITLE
{devel}[gompi/2022a] NVSHMEM v2.5.0

### PR DESCRIPTION
**Pull request by SebastianAchilles**
(created using `eb --new-pr`)

Similar to https://github.com/easybuilders/easybuild-easyconfigs/pull/14832 but newer version and new toolchain version.

### CODE REVIEW

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bdw-opensuse-154 - Linux openSUSE Leap 15.4, x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 510.85.02, Python 3.6.15
See https://gist.github.com/bc1dbacdcd2100bbaf8821cb34f9b70e for a full test report.

File: easybuild/easyconfigs/n/NVSHMEM/NVSHMEM-2.5.0-gompi-2022a-CUDA-11.7.0.eb [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+description = """NVSHMEM is a parallel programming interface based on OpenSHMEM that provides 
+efficient and scalable communication for NVIDIA GPU clusters. NVSHMEM creates a 
+global address space for data that spans the memory of multiple GPUs and can be 
+accessed with fine-grained GPU-initiated operations, CPU-initiated operations, 
+and operations on CUDA streams.
+"""
+
+toolchain = {'name': 'gompi', 'version': '2022a'}
+
+# You need to download the source manully from https://developer.nvidia.com/nvshmem-downloads
```
Comment (jfgrimm) [inline easybuild/easyconfigs/n/NVSHMEM/NVSHMEM-2.5.0-gompi-2022a-CUDA-11.7.0.eb:17]: would be nice to use `download_instructions` here

Comment (SebastianAchilles) [reply easybuild/easyconfigs/n/NVSHMEM/NVSHMEM-2.5.0-gompi-2022a-CUDA-11.7.0.eb]: Oh nice! I was not aware about the `download_instructions` options. Thanks for pointing that out. :+1: 

Added in https://github.com/easybuilders/easybuild-easyconfigs/pull/16254/commits/c4c1a9b5ba47224454f3d4eb3502fa2e520eed03

**Comment by SebastianAchilles** [comment]
Test report by @sebastianachilles
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bdw-opensuse-154 - Linux openSUSE Leap 15.4, x86_64, Intel(R) Core(TM) i7-6900K CPU @ 3.20GHz (broadwell), 2 x NVIDIA NVIDIA GeForce GTX 1060 6GB, 510.85.02, Python 3.6.15
See https://gist.github.com/b81d4ee941b75c4713084f97d3be934f for a full test report.

File: easybuild/easyconfigs/n/NVSHMEM/NVSHMEM-2.5.0-gompi-2022a-CUDA-11.7.0.eb (hunk 1 above)
Comment (jfgrimm) [reply easybuild/easyconfigs/n/NVSHMEM/NVSHMEM-2.5.0-gompi-2022a-CUDA-11.7.0.eb]: yeah it's neat, a lot nicer than having to dig through the easyconfigs when you encounter missing sources

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node142.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/ed524f2d84b69f8057916c1d8c04cfce for a full test report.

File: easybuild/easyconfigs/n/NVSHMEM/NVSHMEM-2.5.0-gompi-2022a-CUDA-11.7.0.eb [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+prebuildopts += 'export NVSHMEM_EXAMPLES_BUILDDIR=${NVSHMEM_BUILDDIR}/examples/obj && '
+prebuildopts += 'export NVSHMEM_OTHERTEST_BUILDDIR=${NVSHMEM_BUILDDIR}/othertest/obj && '
+prebuildopts += 'export NVSHMEM_TEST_BUILDDIR=${NVSHMEM_BUILDDIR}/test/obj && '
+prebuildopts += 'export NVSHMEM_PERFTEST_BUILDDIR=${NVSHMEM_BUILDDIR}/perftest/obj && '
+
+prebuildopts += 'export NVSHMEM_PREFIX=%(installdir)s && '
+prebuildopts += 'export NVSHMEM_EXAMPLES_INSTALL=${NVSHMEM_PREFIX}/examples && '
+prebuildopts += 'export NVSHMEM_OTHERTEST_INSTALL=${NVSHMEM_PREFIX}/othertest && '
+prebuildopts += 'export NVSHMEM_PERFTEST_INSTALL=${NVSHMEM_PREFIX}/perftest && '
+prebuildopts += 'export NVSHMEM_TEST_INSTALL=${NVSHMEM_PREFIX}/test && '
```
Comment (jfgrimm) [inline easybuild/easyconfigs/n/NVSHMEM/NVSHMEM-2.5.0-gompi-2022a-CUDA-11.7.0.eb:57]: I'd say something along the lines of this is more readable, but maybe that's just my preference
```suggestion
prebuildopts = 'export %s &&' % ' '.join([
    'NVSHMEM_USE_GDRCOPY=1',
    'GDRCOPY_HOME=${EBROOTGDRCOPY}',

    'MPI_HOME=${EBROOTOPENMPI}',
    'NVSHMEM_MPI_SUPPORT=1',
    'NVSHMEMTEST_USE_MPI_LAUNCHER=1',

    'NCCL_HOME=${EBROOTNCCL}',
    'NVSHMEM_USE_NCCL=1',

    'NVSHMEM_BUILDDIR=%(builddir)s',
    'NVSHMEM_EXAMPLES_BUILDDIR=${NVSHMEM_BUILDDIR}/examples/obj',
    'NVSHMEM_OTHERTEST_BUILDDIR=${NVSHMEM_BUILDDIR}/othertest/obj',
    'NVSHMEM_TEST_BUILDDIR=${NVSHMEM_BUILDDIR}/test/obj',
    'NVSHMEM_PERFTEST_BUILDDIR=${NVSHMEM_BUILDDIR}/perftest/obj',

    'NVSHMEM_PREFIX=%(installdir)s',
    'NVSHMEM_EXAMPLES_INSTALL=${NVSHMEM_PREFIX}/examples',
    'NVSHMEM_OTHERTEST_INSTALL=${NVSHMEM_PREFIX}/othertest',
    'NVSHMEM_PERFTEST_INSTALL=${NVSHMEM_PREFIX}/perftest',
    'NVSHMEM_TEST_INSTALL=${NVSHMEM_PREFIX}/test',
])
```

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
gpu02.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), 1 x NVIDIA Tesla V100-SXM2-32GB, 510.47.03, Python 3.6.8
See https://gist.github.com/8204f3d088f4911dd06de6c27fc4d44e for a full test report.

**Comment by jfgrimm** [comment]
Test report by @jfgrimm
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node086.pri.viking.alces.network - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6138 CPU @ 2.00GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/36779424f9b11b0eb71b7819d7ebaccc for a full test report.

**Review by jfgrimm — APPROVED** [review]
LGTM

**Comment by jfgrimm** [comment]
Going in, thanks @SebastianAchilles!

**Pull request merged**